Packs/Cylance_Protect/Integrations/Cylance_Protect_v2/Cylance_Protect_v2.py

```diff
@@ -1341,6 +1345,97 @@ def get_policy_details_request(policy_id):  # pragma: no cover
     return res
 
 
+def create_instaquery():
+    query_args = demisto.args()
+    name = query_args['name']
+    description = query_args['description']
+    artifact = query_args['artifact']
+    match_value_type = query_args['match_value_type']
+    match_values = query_args['match_values'].split(",")
+    match_type = query_args['match_type']
+    zones = "".join(query_args['zone'].split("-")).upper()  # Remove '-' and upper case
+    zone_list = zones.split(",")
+
+    # Process the match value
+    if artifact in match_value_type:
+        value_type = re.findall('(?<=\.).*', match_value_type)[0]  # Remove the artifact prefix
+    else:
+        demisto.error('The value type is not suitable with the selected artifact')
+
+    # Create request
+
+    data = {
+        "name": name,
+        "description": description,
+        "artifact": artifact,
+        "match_value_type": value_type,
+        "match_values": match_values,
+        "case_sensitive": False,
+        "match_type": match_type,
+        "zones": zone_list
+    }
+
+    access_token = get_authentication_token()
+    headers = {
+        'Content-Type': 'application/json',
+        'Authorization': 'Bearer ' + access_token
+    }
+    uri = URI_OPTICS
+    res = api_call(uri=uri, method='post', body=data, headers=headers)
+    if res:
+        # Return results to context and war room
+        results = CommandResults(
+            outputs=res,
+            outputs_prefix='InstaQuery.New',
+            outputs_key_field='id'
+        )
+        return_results(results)
+
+
+def get_instaquery_result():
+    query_id = demisto.args().get('query_id')
+
+    # Create request
+    access_token = get_authentication_token()
+    headers = {
+        'Content-Type': 'application/json',
+        'Authorization': 'Bearer ' + access_token
+    }
+    # Endpoint format /instaqueries/v2/{queryID}/results
+    uri = URI_OPTICS + "/" + query_id + "/results"
+    res = api_call(uri=uri, method='get', headers=headers)
+    if res:
+        # Return results to context and war room
+        results = CommandResults(
+            outputs=res,
+            outputs_prefix='InstaQuery.Results',
+            outputs_key_field='id'
+        )
+        return_results(results)
+
+
+def list_instaquery():
+    page = demisto.args().get('page', '1')
+    page_size = demisto.args().get('page_size')
```

Comment:
Consider adding a default value here and in the `yml` file. Are those arguments required by Cylance API? If not, consider making them not mandatory.

Reply:
Yes I have added default values for these 2 args